### PR TITLE
Modern way of finding Python on a system

### PR DESCRIPTION
**Pull request by gpospelov**
+ Provide a consistent way of finding Python interpreter, Python libs, and NumPy on a system with multiple Python installations.
+ New docker container to use for the build (Ubuntu-focal April 2020: newer CMake, newer Qt).
+ CMake minimum required version now 3.14

Two strange issues fixed:
+  The behavior of Qt5 + Docker  + Appveyr while finding *.qrc files on disk.
+ Strange way of parsing Python_EXECUTABLE variables on Appveyor with the mess in "/" against "\\".

### CODE REVIEW

**Comment by rbeerwerth** [comment]
sweet, working nicely!

File: cmake/bornagain/modules/BornAgainConfiguration.cmake [hunk 1]
```diff
@@ -162,10 +162,6 @@ configure_file(${TEMPLATE_DIR}/thisbornagain.csh.in
 
 if(BUILD_DEBIAN)
     set(CMAKE_INSTALL_PREFIX "/usr")
-    execute_process(COMMAND "${PYTHON_EXECUTABLE}" -c
```
Comment (juanmcloaiza) [inline cmake/bornagain/modules/BornAgainConfiguration.cmake:165]: Why isn't this needed anymore?

Comment (juanmcloaiza) [inline cmake/bornagain/modules/BornAgainConfiguration.cmake:165]: This line was intended to define the variable `PYTHON_SITE_PACKAGES`, which is now dropped and instead, int the rest of the files was replaced by `Python_SITELIB`. Where did you define this new variable? What's the logic behind the new way of defining it?

Comment (gpospelov) [reply cmake/bornagain/modules/BornAgainConfiguration.cmake:165]: This is defined [here](https://cmake.org/cmake/help/git-stage/module/FindPython.html).

Comment (juanmcloaiza) [reply cmake/bornagain/modules/BornAgainConfiguration.cmake:165]: to the point, thank you!

**Pull request merged**